File: src/my_project/easy_problems/from151To200/maximum_ascending_subarray_sum.py
from typing import List 

class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:

        temp_max: int = nums[0]
        count: int = 0
        min_index: int  = 0
        max_index: int = 0

        for i in range(1,len(nums)):
            if nums[i-1] < nums[i]:
                max_index = i
                if sum(nums[min_index:max_index+1]) > temp_max:
                    temp_max = sum(nums[min_index:max_index+1])
            else: 
                min_index = i
                max_index = i
        return temp_max

solution = Solution()

print(solution.maxAscendingSum(nums = [10,20,30,5,10,50]))

print(solution.maxAscendingSum(nums = [10,20,30,40,50]))

print(solution.maxAscendingSum(nums = [12,17,15,13,10,11,12]))

print(solution.maxAscendingSum(nums = [100,10,1]))

File: my_project_typescript/easy_problems_typescript/from51To100/maximumAscendingSubarraySum.ts
function maxAscendingSum(nums: number[]): number {

    let tempMax: number = nums[0]
    let minIndex: number = 0
    let maxIndex: number = 0

    for (let i = 1; i < nums.length; i++){
        if (nums[i-1] < nums[i]){
            maxIndex = i
            if ( nums.slice(minIndex, maxIndex+1).reduce((a,b)=> a+b,0) > tempMax){
                tempMax = nums.slice(minIndex, maxIndex+1).reduce((a,b)=> a+b,0) 
            }
        }
        else{
            minIndex = i
            maxIndex = i
        }
    }

    return tempMax

};

console.log(maxAscendingSum([10,20,30,5,10,50]))

console.log(maxAscendingSum([10,20,30,40,50]))

console.log(maxAscendingSum([12,17,15,13,10,11,12]))

console.log(maxAscendingSum([100,10,1]))


console.log([100,1,10].reduce((a,b) => a+b, 0))


